tests/Sdk/Unit/Infrastructure/Service/TaskSet/TaskFromTaskSetBuilderTest.php

```diff
@@ -0,0 +1,201 @@
+            $this->createTaskSetOverrideMapFactoryMock(),
+        );
+
+        // Act
+        $task = $taskSetBuilder->buildTaskFromTaskSet($taskSet, $existingTasks);
+
+        // Assert
+        $this->assertSame($command, $task->getCommands()[0]);
+        $this->assertSame($placeholder, $task->getPlaceholders()[0]);
+        $this->assertSame('taskSetId', $task->getId());
+    }
+
+    /**
+     * @return void
+     */
+    public function testThrowsExceptionTaskWhenSubTaskIdIsNotFound(): void
+    {
+        // Arrange
+        $command = $this->createCommandMock();
+        $placeholder = $this->createPlaceholderMock();
+        $existingTasks = ['NotExistentTaskId' => $this->createTaskMock('taskId', $command, $placeholder)];
+        $taskSet = $this->createTaskSetMock('taskSetId', 'taskId', null);
+        $taskSetBuilder = new TaskFromTaskSetBuilder(
+            $this->createTaskSetPlaceholdersBuilderMock($placeholder),
+            $this->createTaskSetCommandsBuilderMock($command),
+            $this->createTaskSetOverrideMapFactoryMock(),
+        );
+
+        // Act
+        $this->expectException(InvalidArgumentException::class);
+        $taskSetBuilder->buildTaskFromTaskSet($taskSet, $existingTasks);
+    }
+
+    /**
+     * @return void
+     */
+    public function testBuildsTaskWhenSubTaskIsObject(): void
+    {
+        // Arrange
+        $command = $this->createCommandMock();
+        $placeholder = $this->createPlaceholderMock();
+        $task = $this->createTaskMock('taskId', $command, $placeholder);
+        $existingTasks = ['taskId' => $task];
+        $taskSet = $this->createTaskSetMock('taskSetId', null, $task);
+        $taskSetBuilder = new TaskFromTaskSetBuilder(
+            $this->createTaskSetPlaceholdersBuilderMock($placeholder),
+            $this->createTaskSetCommandsBuilderMock($command),
+            $this->createTaskSetOverrideMapFactoryMock(),
+        );
+
+        // Act
+        $task = $taskSetBuilder->buildTaskFromTaskSet($taskSet, $existingTasks);
+
+        // Assert
+        $this->assertSame($command, $task->getCommands()[0]);
+        $this->assertSame($placeholder, $task->getPlaceholders()[0]);
+        $this->assertSame('taskSetId', $task->getId());
+    }
+
+    /**
+     * @param string $taskSetId
+     * @param string|null $taskId
+     * @param \SprykerSdk\SdkContracts\Entity\TaskInterface|null $task
+     *
+     * @return \SprykerSdk\SdkContracts\Entity\TaskSetInterface
+     */
+    protected function createTaskSetMock(string $taskSetId, ?string $taskId, ?TaskInterface $task): TaskSetInterface
+    {
+        $taskMock = $this->createMock(TaskSetInterface::class);
+
+        if ($taskId !== null) {
+            $taskMock->method('getSubTasks')->willReturn([$taskId]);
+        } else {
+            $taskMock->method('getSubTasks')->willReturn([$task]);
+        }
+
+        $taskMock->method('getId')->willReturn($taskSetId);
```

Comment:
Do you really need mock here? Could you use prepared object with data instead?

Reply:
Mocks looks better than anonymous classes in tests. But there is no difference actually.